gateway/api_definition_test.go

```diff
@@ -1037,3 +1038,37 @@ func TestAPIDefinitionLoader_Template(t *testing.T) {
 		executeAndAssert(t, temp)
 	})
 }
+
+func TestAPIExpiration(t *testing.T) {
+	ts := StartTest(nil)
+	defer ts.Close()
+
+	api := BuildAPI(func(spec *APISpec) {
+		spec.Proxy.ListenPath = "/"
+		spec.UseKeylessAccess = true
+		spec.VersionData.NotVersioned = true
+		spec.VersionDefinition.Enabled = true
+	})[0]
+
+	for _, versioned := range []bool{false, true} {
+		api.VersionDefinition.Enabled = versioned
+
+		t.Run(fmt.Sprintf("versioning=%v", versioned), func(t *testing.T) {
+			t.Run("not expired", func(t *testing.T) {
```

Comment:
@furkansenharputlu I still think we should provide more details for the tests, I would like to be able to read the description to understand what's being tested